### PR TITLE
Make reconnecting to a lost RabbitMQ work.

### PR DESCRIPTION
**Pull request by timstoop**
This should solve issue #346 (which I created myself). I'm not sure why nobody else had this issue, maybe they do not upgrade their RabbitMQ very often! Or it might be because I'm using pika 0.10.0. In any case, this solves the issue. Should I increase the requirement for pika to >=0.10.0 as well?

### CODE REVIEW

File: beaver/transports/rabbitmq_transport.py [hunk 1]
```diff
@@ -93,7 +94,7 @@ def _on_connection_open_error(self,non_used_connection=None,error=None):
 
     def _on_connection_closed(self, connection, reply_code, reply_text):
         self._channel = None
-        if self._connection._closing:
+        if self._connection.is_closing:
```
Comment (josegonzalez) [inline beaver/transports/rabbitmq_transport.py]: Probably worth supporting both old and new versions of pika. Pretty easy to check which version in the init of the transport and then just use that one here.

**Comment by timstoop** [comment]
Here I got to admit that I'm not an actual Python dev, just a dabbler in the art. I have no idea what the best way of doing that would be, but I think I have an alternative... Will push in a second.

**Comment by timstoop** [comment]
I don't understand why the kafka tests fail due to my change :/

**Comment by timstoop** [comment]
Actually, I get the same errors when running './setup.py test' on the current master...

**Pull request merged**